Make ConfigurationManager and ModuleConfiguration no longer singleton…

eflumerf opened the pull request:
…s. Work on getting Application_test working with TEST schema additions to daq-cmake

eflumerf committed:
Make ConfigurationManager and ModuleConfiguration no longer singletons. Work on getting Application_test working with TEST schema additions to daq-cmake

eflumerf commented:
This should not be merged until daq-cmake is tagged with https://github.com/DUNE-DAQ/daq-cmake/pull/110

gcrone commented on CMakeLists.txt:
```diff
@@ -25,7 +25,10 @@ find_package(appdal REQUIRED)
 set(APPFWK_DEPENDENCIES ${CETLIB} ${CETLIB_EXCEPT} ers::ers logging::logging cmdlib::cmdlib rcif::rcif iomanager::iomanager opmonlib::opmonlib nlohmann_json::nlohmann_json pistache_shared
   coredal::coredal appdal::appdal oksdbinterfaces::oksdbinterfaces okssystem::okssystem)
 
-include_directories(${CMAKE_BINARY_DIR}/coredal ${CMAKE_BINARY_DIR}/appdal)
```
Actually, I don't think it should be necessary to have these include_directories with the latest daq-cmake.

eflumerf replied:
Removed, still builds.

gcrone commented on include/appfwk/ModuleConfiguration.hpp:
```diff
@@ -21,43 +21,39 @@
+  std::shared_ptr<ConfigurationManager> m_config_mgr;
+
+public:
+  ModuleConfiguration(std::shared_ptr<ConfigurationManager> mgr);
```
Should this be explicit? I seem to remember being told that constructors with a single argument should be explicit but I can't remember why.

eflumerf replied:
It disables implicit conversions, which is generally a good thing.

gcrone reviewed (approved):
Looks OK to me

eflumerf committed:
Mark single-parameter constructor as explicit

eflumerf committed:
Remove a couple of unneeded things from CMakeLists.txt

eflumerf committed:
Remove DAQModuleHelper as connection references are no longer used in OKS

jcfreeman2 committed:
syncing ; previous commit: Merge: dcabece 5b6a027
Author: eflumerf <61473357+eflumerf@users.noreply.github.com>
Date:   Thu Oct 12 09:05:14 2023 -0500

    Merge pull request #265 from DUNE-DAQ/eflumerf/VersionUpdate

    Update version number

eflumerf committed:
Reflect current state of classes in class diagram

eflumerf committed:
Normalize naming of variables, add ConfigurationManager accessor to ModuleConfiguration

eflumerf committed:
Update m_oks_config_spec variable name

jcfreeman2 committed:
syncing ; previous commit: Author: John Freeman <jcfree@fnal.gov>
Date:   Tue Dec 12 19:37:49 2023 -0600

    syncing ; previous commit: Merge: dcabece 5b6a027
    Author: eflumerf <61473357+eflumerf@users.noreply.github.com>
    Date:   Thu Oct 12 09:05:14 2023 -0500

        Merge pull request #265 from DUNE-DAQ/eflumerf/VersionUpdate

        Update version number

eflumerf committed:
Decompress appfwk.drawio

eflumerf committed:
Merge remote-tracking branch 'origin/develop' into eflumerf/NoOKSSingletons

eflumerf committed:
Make init pure virtual

eflumerf committed:
Merge branch 'feature/OKS' into eflumerf/NoOKSSingletons

eflumerf committed:
OKS-ify the DAQModuleManager_test. Doesn't make sense to test the queues and connections here, since DAQModuleManager doesn't provide any information about them

glehmannmiotto committed:
updated gitignore

glehmannmiotto committed:
Merge branch 'eflumerf/NoOKSSingletons' of https://github.com/DUNE-DAQ/appfwk into eflumerf/NoOKSSingletons

eflumerf committed:
Add ERS issue for when a connection expected by a DAQ Module does not exist
